routes/products.js

```diff
@@ -1,19 +1,19 @@
 const express = require('express');
 const router = express.Router();
-const products = require('../src/data/products');
-const categories = require('../src/data/categories');
+const dataService = require('../services/dataService');
 
+/*
+ * This method returns the complete products array
+ */
 router.get('/all', function(req, res, next) {
-    const response = [];
-    products.products.forEach( product => {
-        categories.categories.forEach( category => {
-            if (product.categoryId === category.id) {
-                product.category = category;
-            }
-        });
-        response.push(product);
-    });
-    res.send(response);
+  const products = dataService.getCombinedProductMap();
```

Comment:
Can't we just send this for this response rather than making an array again?

Reply:
f4ww4z said that it has to be an array. Like in the react task 3

Reply:
> f4ww4z said that it has to be an array. Like in the react task 3

@Goutam192002 Can you clarify where I said that? You should send a map (in the form of `Object`) as the response, like Suthagar said. Response should be in the format of:


```json
{
  "P001": {
    "name": "...",
    "categoryId": "...",
    ...
  },
  "P002": {
    "name": "...",
    "categoryId": "...",
    ...
  }
}
```